src/main/java/com/mineinabyss/mobzy/ecs/events/MobzyEventListener.kt

```diff
@@ -0,0 +1,29 @@
+package com.mineinabyss.mobzy.ecs.events
+
+import com.destroystokyo.paper.event.entity.ProjectileCollideEvent
+import com.mineinabyss.geary.ecs.components.Target
+import com.mineinabyss.geary.ecs.components.addComponent
+import com.mineinabyss.geary.minecraft.events.event
+import com.mineinabyss.geary.minecraft.store.geary
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.entity.EntityDamageByEntityEvent
+
+object MobzyEventListener : Listener {
+    @EventHandler
+    fun ProjectileCollideEvent.onCollision() {
+        //TODO: When do we remove this component?
+        geary(entity)?.addComponent(Target(geary(collidedWith) ?: return))
+
+        event(geary(entity), "projectileHit")
+    }
+
+
+    @EventHandler
+    fun EntityDamageByEntityEvent.onDamage() {
+        //TODO: When do we remove this component?
+        geary(damager)?.addComponent(Target(geary(entity) ?: return))
```

Comment:
Should Target components be removed at some point?

Reply:
Could be removed after the `event()` was run, since we're guaranteed all actions have finished processing by then. Since we aren't using spigot event system for this, I think we shouldn't worry too much about the event priority or "what if other events want to use the Target component" since all the events that are ecs specific should be running as GearyActions.

That being said, it may be interesting to design some events that run first, add the appropriate components, then allow other spigot events to use them. You'd then likely just remove the component next tick when the event has finished running (or could add another event listener with the highest priority that removes the component in the end.

I think it doesn't make sense to do this right now since things relating to our ECS should be integrated into our ECS imo.